Automattic/simplenote-android
# Add collaboration UI

Fixes #1479 

### Fix

This PR adds a UI for adding/removing collaborators to a note. Sorry for the size of the PR but I couldn't break more the changes to bring collaboration to Android (they already broken in 4 different issues/PRs https://github.com/Automattic/simplenote-android/issues/1473). 

**Note:** I have stressed tested the UI by adding and removing collaborators as fast as I could and there are some cases in which collaborators are deleted from the list but then added. I presume that this is mostly due to a race condition on the `Simperium` library. I will track the issue in https://github.com/Simperium/simperium-android/issues/238.

### Test

1. Go to a note
2. Tap on the menu on the top right corner (dot ...)
3. Tap on collaborators
4. Tap on Add Collaborator and type anything that is not a valid email. A message invalid collaborator should be shown.
5. Write a valid email address and tap on  `Accept`. Should update the layout to list of collaborators
6. Tap on the cross icon to remove the collaborator. Should update the layout to show the initial message
7. Tap on Add Collaborator and enter an email address from another valid simplenote account
8. The note should be shared to the other account

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.


### Release
